mne/decoding/csp.py

```diff
@@ -129,8 +129,13 @@ def _fit(self, cov_a, cov_b):
         w_b = np.dot(np.dot(p, cov_b), p.T)
         # and solve it
         vals, vecs = linalg.eigh(w_a, w_b)
-        # sort vectors by discriminative power using eigen values
-        ind = np.argsort(np.maximum(vals, 1. / vals))[::-1]
+        # sort vectors by discriminative power using eigenvalues
+        ind = np.argsort(vals)[::-1]
+        vecs = vecs[:, ind]
+        # re-order (first, last, second, second last, third, ...)
+        ind = np.empty_like(ind)
```

Comment:
No need to reallocate `ind`, because it is supposed to be completely filled below.